### PR TITLE
[BlockchainApi] Fix exchange transactions

### LINKED ISSUE #2509: [BlockchainApi] Delete code related to the now deprecated `events` query
### Expected Behavior

Code related to the `events` query in blockchain-api should be removed after the new wallet using `tokenTransactions` has been published for the redeployed Alfajores network.

### Current Behavior

`events` query in blockchain-api has been deprecated in favour of the new `tokenTransactions` query. See #2446

### LINKED ISSUE #3395: Exchange transaction activity is not  shown on “Gold & Wallet tab” when the user Buy or Sell the gold.
**Frequency:** 100%
**App version:** 1.7.0 (104894304)
**Issue Does not repro on** Play store build 1.6.2
**Repro on device:** Google pixel Xl (8.1) 
**Testing Account:** 518 553 1218/[backup Key](https://docs.google.com/spreadsheets/d/1Iy41GGA6e2pES2XHjBn7nb8X3n8zZsu7EW1jvwD2e70/edit?usp=sharing)

**Repro Steps:** 
1) Launch the app and Login with test account
2) Navigate to the Gold tab. 
3) Tap on the Buy button. 
4) Enter an amount in the Amount (Gold) field. 
5) Tap on the Review button. 
6) Tap on Buy # Gold button. 
7) Observed

**Current Behavior:** Exchange activity is not shown. 
**Expected Behavior:** Exchange activity should be shown
**Investigation:** 
- Dollar and Gold amount is shown updated after performing exchange activity. 
- Dollar payment activity shown correct on Wallet tab when user make payment. 

**Impact:** User will never know about previous exchange transactions. 
**Attachment:**[Android_ExchangeActivityIssue.mp4](https://drive.google.com/file/d/1mJsw0lG701lOrmKIKCwDwseosvIwKzV7/view?usp=sharing) (Bug observed at 46 sec)

### PR DESCRIPTION
**Pull request by jeanregisser**
## Description

This PR restores functionality for the wallet activity feed regarding exchange transactions.

The existing API we used broke because there are now events emitted when paying for fees in cUSD. See discussion in https://celo-org.slack.com/archives/CMX4JFMQW/p1586526953169900

A new API has been added to blockscout in order to get all cUSD and cGLD transfers happening in transactions.
This work has been done by @mrsmkl in https://github.com/celo-org/blockscout/pull/102 which has already been deployed on alfajores.

For transactions paying for fees in cUSD we try to detect transfers related to fees (validator, gateway fee recipient, community fund). 
This has risks of breaking, so I've put assertions so we can detect if it happens in the future.

## Other changes

- Removed obsolete feed events query (had been deprecated)

## Tested

Deployed on alfajores and tested successfully.

## Related issues

- Fixes #3395
- Fixed #2509

## Backwards compatibility

Yes

### CODE REVIEW

File: packages/blockchain-api/src/blockscout.ts [hunk 1]
```diff
@@ -129,259 +176,149 @@ export class BlockscoutAPI extends RESTDataSource {
...
+                timestamp,
+              },
+              address,
+              comment,
+              hash,
+            })
+          }
+          break
+
+        case 2: // Exchange events have two corresponding transfers (in and out)
```
Comment (jmrossy) [inline packages/blockchain-api/src/blockscout.ts:330]: Could this large function be broken apart? I'm not sure having a big chunk of code in a switch case is best for readability

File: packages/blockchain-api/src/blockscout.ts [hunk 2]
```diff
@@ -129,259 +176,149 @@ export class BlockscoutAPI extends RESTDataSource {
...
-            currencyCode: event.tokenSymbol,
-            timestamp,
-          },
-          address,
-          comment,
-          hash: txhash,
-        })
+        default:
+          // Warn if anything else happens
+          console.warn(`Unhandled transfers for tx ${transferTx.transactionHash}`)
```
Comment (jmrossy) [inline packages/blockchain-api/src/blockscout.ts]: Nit: maybe error here? We should probably set up alerts for this. Having an error code enum defined somewhere would help with that

**Review by jmrossy — APPROVED** [review]
LGTM, two optional comments below

File: packages/blockchain-api/src/blockscout.ts (hunk 1 above)
Comment (jeanregisser) [reply packages/blockchain-api/src/blockscout.ts:330]: Yep done 👍

File: packages/blockchain-api/src/blockscout.ts (hunk 2 above)
Comment (jeanregisser) [reply packages/blockchain-api/src/blockscout.ts]: Yes!

**Pull request merged**